control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -818,10 +822,36 @@ func (r *HostedControlPlaneReconciler) reconcileAPIServerService(ctx context.Con
 		return fmt.Errorf("failed to reconcile API server service: %w", err)
 	}
 
-	if util.IsPrivateHCP(hcp) {
+	if serviceStrategy.Type == hyperv1.Route {
```

Comment:
nit: I'd find easier to read and understand all the possible casuistic with a swtich case for `serviceStrategy.Type` here, then in default with `util.IsPrivateHCP(hcp) {...`

Reply:
The issue is that we have three logical branches:
* Uses hyperv1. Router: Always create internal route.
    * subbranch is public: Create public route
* Uses hyperv1.LB and is private: Create private LB

The `hyperv1.LB` for the last condition is implicit, because private AWS clusters can only have route or LB apiserver expose strategy. I've added it explicitly to the condition to make that more obvious, the check doesn't really cost anything.